spec/models/customer_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe Customer, type: :model do
+  subject { Customer.new(first_name: "Jack", last_name: "Smith", phone: "8889995678", email: "jsmith@sample.com" )}
+  it "is valid with valid attributes" do
+    expect(subject).to be_valid
+  end
+  it "is not valid without a first_name" do
+    subject.first_name=nil
+    expect(subject).to_not be_valid
+  end
+  it "is not valid without a last_name" do
+    subject.last_name=nil
+    expect(subject).to_not be_valid
+  end
+  it "is not valid without a phone number" do
+    subject.phone=nil
+    expect(subject).to_not be_valid
+  end
+  it "is not valid without an email" do
+    subject.email=nil
+    expect(subject).to_not be_valid
+  end
+  it "is not valid if the phone number is not 10 chars" do
+    if subject.phone.length != 10 
+      expect(subject).to_not be_valid
+    else 
+      expect(subject).to be_valid
+    end
+  end
+  it "is not valid if the phone number is not all digits" do
+    expect(subject.phone).to match(/\d/)
+    expect(subject).to be_valid
+  end
+  it "is not valid if the email address doesn't have a @" do
+    expect(subject.email).to include("@")
+  end
+  it "returns the correct full_name" do
+    expect(subject.full_name).to eq("Jack Smith")
```

Comment:
Rather than hard coding "Jack Smith", it might be better to use string interpolation with the first_name and last_name attributes
`"#{subject.first_name} #{subject.last_name}"`
That way if you decide to change the name later the test will still pass.